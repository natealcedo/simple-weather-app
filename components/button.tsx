"use client";

import React from "react";

import { cn } from "@/lib/utils";

interface ButtonProps {
  onClick?: () => void;
  children?: React.ReactNode;
  className?: string;
}

export function Button(props: ButtonProps) {
  return (
    <button
      className={cn(
        "flex-1 rounded-full bg-gray-700 px-6 py-4 text-3xl font-bold text-white hover:bg-gray-500",
        props.className
      )}
      onClick={() => {
        if (props.onClick) props.onClick();
      }}
    >
      {props.children}
    </button>
  );
}
