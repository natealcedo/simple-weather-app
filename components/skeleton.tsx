import { cn } from "@/lib/utils";

export function Skeleton({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) {
  return <div className={cn("bg-muted animate-pulse rounded-md", className)} {...props} />;
}
