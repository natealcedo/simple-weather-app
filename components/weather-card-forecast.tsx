import React from "react";

import { WeatherForecastItem } from "@/app/forecast/forecast";

interface CardProps {
  weatherForeCast: WeatherForecastItem;
  additionalClassNames?: string;
}

export function WeatherCardForeCast(props: CardProps) {
  return (
    <div
      className={`max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md ${props.additionalClassNames}`}
    >
      <div className="p-4">
        <h2 className="mb-2 text-xl font-semibold text-gray-700">{props.weatherForeCast.date}</h2>
        <p className="text-gray-700">{props.weatherForeCast.prediction}</p>
      </div>
    </div>
  );
}
