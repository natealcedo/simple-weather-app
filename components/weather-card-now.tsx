import React from "react";

import { WeatherData } from "@/app/now/page";

interface CardProps {
  weatherData: WeatherData["items"][number];
  additionalClassNames?: string;
}

export function WeatherCardNow(props: CardProps) {
  return (
    <div
      className={`max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md ${props.additionalClassNames}`}
    >
      <div className="p-4">
        <h2 className="mb-2 text-xl font-semibold text-gray-700">{props.weatherData.area}</h2>
        <p className="text-gray-700">{props.weatherData.forecast}</p>
      </div>
    </div>
  );
}
