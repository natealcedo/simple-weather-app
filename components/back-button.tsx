"use client";

import React from "react";
import Link from "next/link";
import { usePathname } from "next/navigation";

import { Button } from "@/components/button";

export function BackButton() {
  const pathName = usePathname();
  // Only render the back button if the current path is now or forecast
  return (
    (pathName === "/now" || pathName === "/forecast") && (
      <div className="absolute left-4 top-4">
        <Link href="/">
          <Button>Back</Button>
        </Link>
      </div>
    )
  );
}
