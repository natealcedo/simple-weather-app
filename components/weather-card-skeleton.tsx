import { Skeleton } from "@/components/skeleton";

export function WeatherCardSkeleton() {
  return (
    <div className="flex gap-4">
      {[1, 2, 3, 4].map((index) => (
        <Skeleton
          key={index}
          className={`h h-[100px] w-[240px] max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md`}
        >
          <Skeleton className="flex flex-col gap-4 p-4">
            <Skeleton className="mb-2 h-4 w-full bg-gray-700"></Skeleton>
            <Skeleton className="h-6 w-full bg-gray-700" />
          </Skeleton>
        </Skeleton>
      ))}
    </div>
  );
}
