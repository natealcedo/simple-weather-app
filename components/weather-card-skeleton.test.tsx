import React from "react";
import { render } from "@testing-library/react";

import { WeatherCardSkeleton } from "@/components/weather-card-skeleton";

describe("WeatherCardSkeleton", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<WeatherCardSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
