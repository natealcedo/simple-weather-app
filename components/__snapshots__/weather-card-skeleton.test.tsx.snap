// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardSkeleton matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex gap-4"
  >
    <div
      class="animate-pulse h h-[100px] w-[240px] max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md"
    >
      <div
        class="bg-muted animate-pulse rounded-md flex flex-col gap-4 p-4"
      >
        <div
          class="animate-pulse rounded-md mb-2 h-4 w-full bg-gray-700"
        />
        <div
          class="animate-pulse rounded-md h-6 w-full bg-gray-700"
        />
      </div>
    </div>
    <div
      class="animate-pulse h h-[100px] w-[240px] max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md"
    >
      <div
        class="bg-muted animate-pulse rounded-md flex flex-col gap-4 p-4"
      >
        <div
          class="animate-pulse rounded-md mb-2 h-4 w-full bg-gray-700"
        />
        <div
          class="animate-pulse rounded-md h-6 w-full bg-gray-700"
        />
      </div>
    </div>
    <div
      class="animate-pulse h h-[100px] w-[240px] max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md"
    >
      <div
        class="bg-muted animate-pulse rounded-md flex flex-col gap-4 p-4"
      >
        <div
          class="animate-pulse rounded-md mb-2 h-4 w-full bg-gray-700"
        />
        <div
          class="animate-pulse rounded-md h-6 w-full bg-gray-700"
        />
      </div>
    </div>
    <div
      class="animate-pulse h h-[100px] w-[240px] max-w-sm overflow-hidden rounded-lg bg-gray-300 opacity-80 shadow-md"
    >
      <div
        class="bg-muted animate-pulse rounded-md flex flex-col gap-4 p-4"
      >
        <div
          class="animate-pulse rounded-md mb-2 h-4 w-full bg-gray-700"
        />
        <div
          class="animate-pulse rounded-md h-6 w-full bg-gray-700"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
